src/plugins/discover/public/application/embeddable/view_saved_search_action.ts

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+import { ActionExecutionContext } from 'src/plugins/ui_actions/public';
+import { ApplicationStart } from 'kibana/public';
+import { i18n } from '@kbn/i18n';
+import { IEmbeddable, ViewMode } from '../../../../embeddable/public';
+import { Action } from '../../../../ui_actions/public';
+import { SavedSearchEmbeddable } from './saved_search_embeddable';
+import { SEARCH_EMBEDDABLE_TYPE } from '../../../common';
+
+export const ACTION_VIEW_SAVED_SEARCH = 'ACTION_VIEW_SAVED_SEARCH';
+
+export interface ViewSearchContext {
+  embeddable: IEmbeddable;
+}
+
+export class ViewSavedSearchAction implements Action<ViewSearchContext> {
+  public id = ACTION_VIEW_SAVED_SEARCH;
+  public readonly type = ACTION_VIEW_SAVED_SEARCH;
+
+  constructor(private readonly application: ApplicationStart) {}
+
+  async execute(context: ActionExecutionContext<ViewSearchContext>): Promise<void> {
+    const { embeddable } = context;
+    const savedSearchId = (embeddable as SavedSearchEmbeddable).getSavedSearch().id;
+    const path = `#/view/${encodeURIComponent(savedSearchId)}`;
+    const app = embeddable ? embeddable.getOutput().editApp : undefined;
+    await this.application.navigateToApp(app ? app : 'discover', { path });
+  }
+
+  getDisplayName(context: ActionExecutionContext<ViewSearchContext>): string {
+    return i18n.translate('discover.savedSearchEmbeddable.action.viewSavedSearch.displayName', {
+      defaultMessage: 'View saved search',
```

Comment:
I'd tend more towards using a name like "Open in Discover", which uses - what I feel - are a bit more common terms across Kibana users, than what a "saved search" is. @kertal getting another thought about that?

Reply:
Edit operation is named "Edit saved search" though.

Reply:
How does

`Open in Discover`

sound for both operations. It's simple.

Agree that saved searches are not that well known also they are often confused with saved queries, something we maybe should also discuss ... but not here

Reply:
on the other hand "Edit saved search" is like "Edit lens", and displayed on a different level of this menu. 
Think this should be kept ... but `Open in Discover` is fine for the new addition IMO.

Reply:
The problem is that the name of the "Edit" operation is automatically generated from the type of the saved object by prefixing it with "Edit". So we would need a bit more changes if we want to rename that too.

Matthias, not sure I understand your last comment correctly: So you'd be for renaming this to "Open in Discover" or against?

Reply:
I'm for renaming!